content/en/blog/_posts/2021-12-15-StatefulSet-PVC-Auto-Deletion.md

```diff
@@ -0,0 +1,52 @@
+PVCs are not touched, or `Delete`, which means that PVCs are deleted. The deletion is done with
+a normal [object deletion](/docs/concepts/architecture/garbage-collection/), so that, for example,
+all retention policies for the underlying PV are respected.
+
+This policy forms a matrix with four cases. I’ll walk through and give an example for each one.
+
+  * **whenDeleted and whenScaled are both Retain.** This matches the existing behavior for StatefulSets, where no PVCs are deleted. This is also the default retention policy. It’s appropriate to use when data on StatefulSet volumes may be irreplaceable and should only be deleted manually.
+
+  * **whenDeleted is Delete and whenScaled is Retain.** In this case, PVCs are deleted only when the entire StatefulSet is deleted. If the StatefulSet is scaled down, PVCs are not touched, meaning they are available to be reattached if a scale-up occurs with any data from the previous replica. This might be used for a temporary StatefulSet, such as in a CI instance or ETL pipeline, where the data on the StatefulSet is needed only during the lifetime of the StatefulSet lifetime, but while the task is running the data is not easily reconstructible. Any retained state is needed for any replicas that scale down and then up.
+
+  * **whenDeleted and whenScaled are both Delete.** PVCs are deleted immediately when their replica is no longer needed. Note this does not include when a Pod is deleted and a new version rescheduled, for example when a node is drained and Pods need to migrate elsewhere. The PVC is deleted only when the replica is no longer needed as signified by a scale-down or StatefulSet deletion. This use case is for when data does not need to live beyond the life of its replica. Perhaps the data is easily reconstructable and the cost savings of deleting unused PVCs is more important than quick scale-up, or perhaps that when a new replica is created, any data from a previous replica is not usable and must be reconstructed anyway.
+
+  * **whenDeleted is Retain and whenScaled is Delete.** This is similar to the previous case, when there is little benefit to keeping PVCs for fast reuse during scale-up. An example of a situation where you might use this is an Elasticsearch cluster.. Typically you would scale that workload up and down to match demand, whilst ensuring a minimum number of replicas (for example: 3). When scaling down, data is migrated away from removed replicas and there is no benefit to retaining those PVCs. However, it can be useful to bring the entire Elasticsearch cluster down temporarily for maintenance. If you need to take the Elasticsearch system offline, you can do this by temporarily deleting the StatefulSet, and then bringing the Elasticsearch cluster back by recreating the StatefulSet. The PVCs holding the Elasticsearch data will still exist and the new replicas will automatically use them.
```

Comment:
Within the bold bits, I'd be tempted to put `whenDeleted` and `whenScale` in backticks (like I did there). I think it'd actually be easier to read like that.

Also, a nit (I think it might actually be my typo from a previous suggestion):
```markdown
this is an Elasticsearch cluster.. Typically
```
two periods

Reply:
This will also be easier on localization teams if you wrap these at 80 to 100 character length lines, rather than one paragraph per line.

Reply:
both done.

I also put `Retain` and `Delete` in backticks which is consistent what I've done in the text above.